nx/include/switch/kernel/rwlock.h

```diff
@@ -6,12 +6,15 @@
  */
 #pragma once
 #include "../kernel/mutex.h"
+#include "../kernel/condvar.h"
 
 /// Read/write lock structure.
 typedef struct {
-    RMutex r;
-    RMutex g;
-    u64 b;
+    Mutex mutex;
+    CondVar condvar_readers;
+    CondVar condvar_writer;
+    u64  readers;
+    bool writer;
```

Comment:
Save space and avoid alignment padding by using bitfields:
```c
    u32 readers : 31;
    bool writer : 1;
```

Reply:
fixed